User story#2

pmarx92 opened the pull request:
I added a form which takes the name, weight, length, time, date, and location as input.
Added a localstorage aswell, you can check if the localstorage works with the button i added to the page. 
It sends a console.log().

#2

s-kond commented:
> I added a form which takes the name, weight, length, time, date, and location as input. Added a localstorage aswell, you can check if the localstorage works with the button i added to the page. It sends a console.log().

Kleiner Trick, den ich von Marc gelernt habe: du kannst einfach deine userstory/issue hier verlinken mit #2

s-kond commented on pages/Add/index.js:
```diff
@@ -1,7 +1,131 @@
 export default function Add() {
+  const [startDate, setStartDate] = useState(new Date());
+  const [fishList, setFishList] = useState(
+    loadLocalStorage("localfFishList") ?? []
```
little typo: "localfFishList" -> "localFishList"
same typo in line 18

s-kond commented on pages/Add/index.js:
```diff
@@ -1,7 +1,131 @@
+
+  useEffect(() => {
+    setLocalStorage("localfFishList", fishList);
+  });
```
I think you have to add empty square-brackets to line 19, so that this useEffect takes places only once the app.js is mounted

```suggestion
  },[]);
```

VladF146 replied:
An empty dependency array updates your localStorage once and then never again. But if you want to update it whenever your fishList updates, you should put fishList inside the array :)

s-kond commented on pages/Add/index.js:
```diff
@@ -1,7 +1,131 @@
+      location: e.target.location.value,
+      date: startDate,
+    };
+    setFishList([...fishList].concat(newFish));
```
Didn't know you could do it with 'concat' - I would have done it like this: 
```suggestion
    setFishList([...fishList, newFish]);
```

pmarx92 replied:
Yeah i like the concat on this one because it is more intuitive to me. I can remember the syntax far easier.
But of course does your solution work aswell :)

pmarx92 commented:
> > I added a form which takes the name, weight, length, time, date, and location as input. Added a localstorage aswell, you can check if the localstorage works with the button i added to the page. It sends a console.log().
> 
> Kleiner Trick, den ich von Marc gelernt habe: du kannst einfach deine userstory/issue hier verlinken mit #2 - das spart Schreibarbeit. Und wenn du 'fixes #2' schreibst, müsste der das auf dem Kanban-Board/Projekt sogar automatisch nach 'Done' verschieben, sobald du diesen PR merged.

Ach cool, wusste nicht das sowas geht. Danke :)
